checks/frozen_deps.go

```diff
@@ -188,40 +213,80 @@ func validateDockerfile(path string, content []byte,
 	return ret, nil
 }
 
-// ============================================================
-// ===================== Github workflows =====================
-// ============================================================.
+func isGitHubWorkflowScriptFreeOfInsecureDownloads(c *checker.CheckRequest) checker.CheckResult {
+	return CheckFilesContent(CheckFrozenDeps, "*", false, c, validateGitHubWorkflowShellScriptDownloads)
+}
+
+func validateGitHubWorkflowShellScriptDownloads(pathfn string, content []byte,
+	logf func(s string, f ...interface{})) (bool, error) {
+	if len(content) == 0 {
+		return false, ErrEmptyFile
+	}
+
+	var workflow gitHubActionWorkflowConfig
+	err := yaml.Unmarshal(content, &workflow)
+	if err != nil {
+		return false, fmt.Errorf("!! frozen-deps - cannot unmarshal file %v\n%v: %w", pathfn, string(content), err)
+	}
+
+	githubVarRegex := regexp.MustCompile(`{{[^{}]*}}`)
+	validated := true
+	// https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell.
+	defaultShell := "bash"
+	scriptContent := ""
+	for _, job := range workflow.Jobs {
+		if job.Defaults.Run.Shell != "" {
+			defaultShell = job.Defaults.Run.Shell
+		}
+
+		for i, step := range job.Steps {
+			if step.Run == "" {
+				continue
+			}
+
+			shell := defaultShell
```

Comment:
i think we can simplify this by not having this `defaultShell` var.

Just something like:

```go
if step.Shell != "" && !isShell(step.Shell) {
  continue
}
```

since the default is bash, we only need to check if the user specified shell is valid?

Reply:
there's an option to change the default shell for all runs in the workflow, see https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_iddefaults.
That's why I have this defaultShell variable. Resolving for now, feel free to re-open.